File: solutions/ocaml/017-remove-nth-node-from-end-of-list.ml
(*
  Problem 17: Remove Nth Node From End of List (LeetCode 19)
  Difficulty: Med
  Language: OCaml
*)
let remove_nth_from_end lst n =
  let len = List.length lst in
  let idx = len - n in
  List.filteri (fun i _ -> i <> idx) lst

File: solutions/haskell/017-remove-nth-node-from-end-of-list.hs
{-
  Problem 17: Remove Nth Node From End of List (LeetCode 19)
  Difficulty: Med
  Language: Haskell
-}
removeNthFromEnd :: [a] -> Int -> [a]
removeNthFromEnd xs n =
  let len = length xs
      idx = len - n
  in take idx xs ++ drop (idx + 1) xs
